mlflow/pyspark/ml/__init__.py

```diff
@@ -643,19 +834,73 @@ def fit_mlflow(original, self, *args, **kwargs):
             _logger.warning(_get_warning_msg_for_fit_call_with_a_list_of_params(self))
             return original(self, *args, **kwargs)
         else:
-            estimator = self.copy(params)
+            estimator = self.copy(params) if params is not None else self
```

Comment:
https://spark.apache.org/docs/latest/api/python/_modules/pyspark/ml/param.html#Params.copy

```python
    def copy(self, extra=None):
        """
        Creates a copy of this instance with the same uid and some
        extra params. The default implementation creates a
        shallow copy using :py:func:`copy.copy`, and then copies the
        embedded and extra parameters over and returns the copy.
        Subclasses should override this method if the default approach
        is not sufficient.

        Parameters
        ----------
        extra : dict, optional
            Extra parameters to copy to the new instance

        Returns
        -------
        :py:class:`Params`
            Copy of this instance
        """
        if extra is None:
            extra = dict()
        that = copy.copy(self)
        that._paramMap = {}
        that._defaultParamMap = {}
        return self._copyValues(that, extra)
```

`copy` seems to work for `None` or old versions of pyspark have different behaviors?

Reply:
https://github.com/apache/spark/blame/44b7931936d9eff4d8f3054abdef3363af26afb6/python/pyspark/ml/param/__init__.py#L373

Reply:
yes. but if params is None we don't need do the `copy.copy(self)` step. This is a small optimization.